### ISSUE TITLE
Colmap reconstruction step repetition/poor end result issue

### ISSUE THREAD

**Issue by arv8315** [colmap/colmap#2318]
**Describe the bug**
I have a dataset with high overlap (overall 300+ images when going around object with single camera whose pose was kept fixed. I do not have camera access so cannot collect new sample) features on photos of said object (background has been removed with preprocessing and just black) are clear to human eyes. SFM in colmap does feature extraction and matching ok. I can see several correspondences between image pairs which look ok. During the reconstruction phases, solves up to 100+ camera poses correctly and then throws it away and keeps restarting. After a fair while , 20+ minutes ends up solving just a few poses and says ‘no good initial image pair found’ and stops. The point cloud has a big cavity on one side. I tried to reduce number of min_num_matches and bundle adjustment ‘num_images’ without much more improvement. In one case that worked and it got 3/4 around and atleast got those poses done. In couple of other cases, ‘repeats’ many times and then only get paltry number of poses and relatively ‘incomplete’ point cloud. Any suggestions?

**To Reproduce**
just using the colmap gui and following standard set of steps

**Expected behavior**
Point cloud comes out fine like it did for two cases out of the lot.

**Screenshots**
Will supply that soon. Need to get some interim screenshots to tell ‘story’.

**Environment:**
 - OS: Linux, Ubuntu 20.04
 - COLMAP Version, latest main one from repo

**Comment by arv8315** [comment]
Toy plane final sparse PC:

![image](https://github.com/colmap/colmap/assets/11643561/90a2c0b4-fef5-42f5-bcb2-b09329867482)

Interim PC:
![image](https://github.com/colmap/colmap/assets/11643561/1fe8022c-9cd0-4d82-aa1f-b4af2fa44177)

I have not yet determined why the poses collapse to small number of camera poses and the resulting PC looks much worse than the interim one.

**Comment by tsattler** [comment]
> I have not yet determined why the poses collapse to small number of camera poses and the resulting PC looks much worse than the interim one.

Colmap produces potentially multiple models. Did you check all of them?

**Comment by jaco001** [comment]
First sin: you maked just 1 orbit around model.

If your model have 'thin' side... game over. Colmap don't have imagination to 'glue' two sides of ypur model. But this is not its fault.
What now? (in your case)
Put images with background! If you are lucky - some part of background work like transistion plane and this could give colmap more room to glue parts that it recognize. (background must be sharp). There are few minor things to do, but maybe next session is better and faster thing to do.

Next time do more orbits around model (thru the top and different height at side). If model have thin side put markers around it to help software to catch it.

**Comment by arv8315** [comment]
> > I have not yet determined why the poses collapse to small number of camera poses and the resulting PC looks much worse than the interim one.
> 
> Colmap produces potentially multiple models. Did you check all of them?

yup, I see multiple models from the drop down menu, one of which encircles the whole way. The last one on the list corresponded with newest model. In this case, it is good that one went all the way and can use that for dense PC perhaps, but in case none of the models encircled all the way and need to stitch them together., will check if there is a way.

**Comment by arv8315** [comment]
> First sin: you maked just 1 orbit around model.
> 
> If your model have 'thin' side... game over. Colmap don't have imagination to 'glue' two sides of ypur model. But this is not its fault. What now? (in your case) Put images with background! If you are lucky - some part of background work like transistion plane and this could give colmap more room to glue parts that it recognize. (background must be sharp). There are few minor things to do, but maybe next session is better and faster thing to do.
> 
> Next time do more orbits around model (thru the top and different height at side). If model have thin side put markers around it to help software to catch it.

In my case, I had to remove the background because the camera was actually held flxed while the object was rotated on a turntable. I do not have that setup and just working with post-processing. Perhaps, would have to take the info of each camera pose that has been computed as well as knowledge of number of cameras in a model and then find some paranorma scene and convert it to frames and make each frame background for one of the photos with right overlap when from one frame to the next and then redo? That would glue the models then? Hopefully there is an easier way to glue the models. 


For future data collection, ensure that multiple orbits are collected and camera it the one made to move. However there might be some constraints with performing that.

**Comment by arv8315** [comment]
https://github.com/colmap/colmap/issues/1703 - checking this

**Comment by jaco001** [comment]
In the future:
you can use marker-mat or someting like this 
![image](https://github.com/colmap/colmap/assets/67650086/6bf8c52f-f953-411b-bfa3-847843e51a44)

**Comment by arv8315** [comment]
> Next time do more orbits around model (thru the top and different height at side). If model have thin side put markers around it to help software to catch it.

Okay, by thin model, were you referring to any thin parts in the object being SFM'd? The issue was that it is difficult to segment the turntable as well the object of interest and subtract out the rest of the background. If we have some sequence of frames, then it is cumbersome to have to do the segmentation and removal of unwanted background manually. Fortunately, there appear to be more advanced segmentation tools now. 

If it serves as a 'transition plane' as you say, hopefully will help in other cases where I had so many models that I had to then merge. Merging can only be done pair at a time looks like unless I am mistaken. Hopefully do not get merge failures like I have seen with issues raised by others. Possibly sequencing matter with order of pairs being merged. 


Meanwhile with the turntable, I think I have some random features on the one I have to work with (restricted to post-processing only for current dataset). The random features are not as extensive as the pic you shared but hopefully adequate. Otherwise, back to data collection round 2!

**Comment by ahojnnes** [comment]
Closing: model evaluation question answered by maintainer.
